contracts/solvprotocol/erc-3525/SpanningERC3525Upgradeable.sol

```diff
@@ -254,27 +353,31 @@ contract ERC3525Upgradeable is Initializable, ContextUpgradeable, IERC3525Metada
     }
 
     function tokenOfOwnerByIndex(address owner_, uint256 index_) public view virtual override returns (uint256) {
+        return tokenOfOwnerByIndex(getAddressFromLegacy(owner_), index_);
+    }
+
+    function tokenOfOwnerByIndex(bytes32 owner_, uint256 index_) public view virtual override returns (uint256) {
         require(index_ < ERC3525Upgradeable.balanceOf(owner_), "ERC3525: owner index out of bounds");
         return _addressData[owner_].ownedTokens[index_];
     }
 
-    function _setApprovalForAll(address owner_, address operator_, bool approved_) internal virtual {
+    function _setApprovalForAll(bytes32 owner_, bytes32 operator_, bool approved_) internal virtual {
         require(owner_ != operator_, "ERC3525: approve to caller");
 
         _addressData[owner_].approvals[operator_] = approved_;
 
         emit ApprovalForAll(owner_, operator_, approved_);
```

Comment:
i think these events are defined in the 721 base contract which we switched to spanning.

spanning type is SpanningApprovalForAll

original event uses address instead of bytes32

Reply:
matched to pattern for SpanningERC721 where both event type are emitted

for crosschain events some user data may be partially incomplete to retain backwards compatibility